experimental/integrations/prototype-spark/marquez-spark-agent/src/main/java/marquez/spark/agent/lifecycle/SparkSQLExecutionContext.java

```diff
@@ -1,184 +1,187 @@
+        .run(buildRun(
+            buildLogicalPlanFacet(queryExecution.logical()),
+            buildPhysicalPlanFacet(queryExecution.executedPlan())))
+        .job(buildJob())
+        .eventTime(ZonedDateTime.now())
+        .eventType("START")
+        .producer("spark")
+        .build();
+
+    marquezContext.emit(event);
   }
 
-  private void print(String prefix, SparkPlanInfo i) {
-    extraInfo.append(prefix + i.nodeName()+ "(" + i.simpleString()+ " metadata: " + i.metadata() + " metrics: "+ i.metrics() + ")" ).append("\n");
-    Collection<SparkPlanInfo> children = JavaConversions.asJavaCollection(i.children());
-    for (SparkPlanInfo child : children) {
-      print(prefix + "  ", child);
+  @Override
+  public void end(SparkListenerJobEnd jobEnd) {
+    log.info("Ending job as part of spark-sql:" + jobEnd.jobId());
+    QueryExecution queryExecution = SQLExecution.getQueryExecution(executionId);
+    if (queryExecution == null) {
+      log.info("No execution info {}", queryExecution);
+      return;
+    }
+    if (jobEnd.jobResult() instanceof JobFailed) {
+//      error = ((JobFailed)jobEnd.jobResult()).exception();
     }
+
+    LineageEvent event = LineageEvent.builder()
+        .inputs(buildInputs(queryExecution.logical()))
+        .outputs(buildOutputs(queryExecution.logical()))
+        .run(buildRun(
+            buildLogicalPlanFacet(queryExecution.logical()),
+            buildPhysicalPlanFacet(queryExecution.executedPlan())))
+        .job(buildJob())
+        .eventTime(ZonedDateTime.now())
+        .eventType(getEventType(jobEnd.jobResult()))
+        .producer("spark")
+        .build();
+
+    marquezContext.emit(event);
   }
 
-  private void print(String prefix, SparkPlan executedPlan) {
-    extraInfo.append(prefix + executedPlan.treeString(true, true)).append("\n");
+  protected String getEventType(JobResult jobResult) {
+    if (jobResult.getClass().getSimpleName().startsWith("JobSucceeded")){
+      return "COMPLETE";
+    }
+    return "FAIL";
   }
 
-  private void print(String prefix, LogicalPlan plan) {
-    String extra = "";
-    if (plan instanceof LogicalRelation) {
-      LogicalRelation lr = (LogicalRelation) plan;
+  private List<LineageDataset> buildInputs(LogicalPlan logical) {
+    List<LineageDataset> inputDatasets = new ArrayList<>();
+    Collection<LogicalPlan> leaves = asJavaCollection(logical.collectLeaves());
+    for (LogicalPlan leaf : leaves) {
+      if (!(leaf instanceof LogicalRelation)) {
+        continue;
+      }
+      LogicalRelation lr = (LogicalRelation) leaf;
       BaseRelation lrr = lr.relation();
-      extra = "||| " + lrr + " " + lr.output() + " " + lr.catalogTable() + " " + lr.isStreaming();
-      if (lrr instanceof HadoopFsRelation) {
-        FileIndex location = ((HadoopFsRelation)lrr).location();
-        extra += "|||" + Arrays.toString(location.inputFiles());
+      if (!(lrr instanceof HadoopFsRelation)) {
+        continue;
```

Comment:
We'd probably want to keep track of the unknown leaves since they are most likely inputs we don't know about and saves them to a run facet? Could be some DebuggingFacet or UnknownInputFacet. Not sure if they need to be standard or just custom facets defined here.